From 9be37cc7d50a9f0d0610bb8d0a5867a7f95f8ee2 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 21 Jun 2024 14:39:20 +0200
Subject: [PATCH 0935/1695] dts: stm32: reduce access-controllers-cells to 1 on
 stm32mp25 boards

Limit the number of access-controllers cells to 1

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c363601185e0d91e61e5ac037053f59bf9d616f0 dts: stm32: reduce access-controllers-cells to 1 on stm32mp25 boards]
Change-Id: I157cb9630551d9467004bd927ac7b952a6d85e75
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/389468
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi             | 54 +++++++++----------
 .../arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi | 10 ++--
 .../dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi    | 10 ++--
 core/arch/arm/dts/stm32mp25xc.dtsi            |  6 +--
 core/arch/arm/dts/stm32mp25xf.dtsi            |  6 +--
 5 files changed, 43 insertions(+), 43 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index f6929fbbd..0afbe3af3 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -261,7 +261,7 @@
 			reg = <0x42080000 0x1000>;
 			#address-cells = <1>;
 			#size-cells = <1>;
-			#access-controller-cells = <2>;
+			#access-controller-cells = <1>;
 
 			lptimer1: timer@40090000 {
 				compatible = "st,stm32mp25-lptimer";
@@ -271,7 +271,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM1_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM1_ID>;
 				status = "disabled";
 
 				counter {
@@ -305,7 +305,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM2_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM2_ID>;
 				status = "disabled";
 
 				counter {
@@ -336,7 +336,7 @@
 				reg = <0x400e0000 0x400>;
 				interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART2>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_USART2_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART2_ID>;
 				status = "disabled";
 			};
 
@@ -345,7 +345,7 @@
 				reg = <0x400f0000 0x400>;
 				interrupts = <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART3>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_USART3_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART3_ID>;
 				status = "disabled";
 			};
 
@@ -354,7 +354,7 @@
 				reg = <0x40100000 0x400>;
 				interrupts = <GIC_SPI 126 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART4>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_UART4_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART4_ID>;
 				status = "disabled";
 			};
 
@@ -363,7 +363,7 @@
 				reg = <0x40110000 0x400>;
 				interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART5>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_UART5_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART5_ID>;
 				status = "disabled";
 			};
 
@@ -375,7 +375,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
-				access-controllers = <&rifsc STM32MP25_RIFSC_I2C1_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C1_ID>;
 				status = "disabled";
 			};
 
@@ -387,7 +387,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
-				access-controllers = <&rifsc STM32MP25_RIFSC_I2C2_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C2_ID>;
 				status = "disabled";
 			};
 
@@ -399,7 +399,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
-				access-controllers = <&rifsc STM32MP25_RIFSC_I2C3_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C3_ID>;
 				status = "disabled";
 			};
 
@@ -411,7 +411,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
-				access-controllers = <&rifsc STM32MP25_RIFSC_I2C4_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C4_ID>;
 				status = "disabled";
 			};
 
@@ -423,7 +423,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
-				access-controllers = <&rifsc STM32MP25_RIFSC_I2C5_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C5_ID>;
 				status = "disabled";
 			};
 
@@ -435,7 +435,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
-				access-controllers = <&rifsc STM32MP25_RIFSC_I2C6_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C6_ID>;
 				status = "disabled";
 			};
 
@@ -447,7 +447,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
-				access-controllers = <&rifsc STM32MP25_RIFSC_I2C7_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C7_ID>;
 				status = "disabled";
 			};
 
@@ -456,7 +456,7 @@
 				reg = <0x40220000 0x400>;
 				interrupts = <GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART6>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_USART6_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART6_ID>;
 				status = "disabled";
 			};
 
@@ -465,7 +465,7 @@
 				reg = <0x402c0000 0x400>;
 				interrupts = <GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART9>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_UART9_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART9_ID>;
 				status = "disabled";
 			};
 
@@ -474,7 +474,7 @@
 				reg = <0x40330000 0x400>;
 				interrupts = <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART1>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_USART1_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_USART1_ID>;
 				status = "disabled";
 			};
 
@@ -483,7 +483,7 @@
 				reg = <0x40370000 0x400>;
 				interrupts = <GIC_SPI 148 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART7>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_UART7_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART7_ID>;
 				status = "disabled";
 			};
 
@@ -492,7 +492,7 @@
 				reg = <0x40380000 0x400>;
 				interrupts = <GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART8>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_UART8_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_UART8_ID>;
 				status = "disabled";
 			};
 
@@ -502,7 +502,7 @@
 				interrupts = <GIC_SPI 145 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_HASH>;
 				resets = <&rcc HASH_R>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_HASH_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_HASH_ID>;
 				status = "disabled";
 			};
 
@@ -512,7 +512,7 @@
 				clocks = <&clk_rcbsec>, <&rcc CK_BUS_RNG>;
 				clock-names = "rng_clk", "rng_hclk";
 				resets = <&rcc RNG_R>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_RNG_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_RNG_ID>;
 			};
 
 			iwdg1: watchdog@44010000 {
@@ -521,7 +521,7 @@
 				interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_IWDG1>, <&rcc LSI_CK>;
 				clock-names = "pclk", "lsi";
-				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG1_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG1_ID>;
 				status = "disabled";
 			};
 
@@ -531,7 +531,7 @@
 				interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_IWDG2>, <&rcc LSI_CK>;
 				clock-names = "pclk", "lsi";
-				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG2_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG2_ID>;
 				status = "disabled";
 			};
 
@@ -543,7 +543,7 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				i2c-analog-filter;
-				access-controllers = <&rifsc STM32MP25_RIFSC_I2C8_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_I2C8_ID>;
 				status = "disabled";
 			};
 
@@ -555,7 +555,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM3_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM3_ID>;
 				status = "disabled";
 
 				counter {
@@ -589,7 +589,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM4_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM4_ID>;
 				status = "disabled";
 
 				counter {
@@ -623,7 +623,7 @@
 				clock-names = "mux";
 				#address-cells = <1>;
 				#size-cells = <0>;
-				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM5_ID 0>;
+				access-controllers = <&rifsc STM32MP25_RIFSC_LPTIM5_ID>;
 				status = "disabled";
 
 				counter {
diff --git a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
index ce7a41c5d..0b73b5c2c 100644
--- a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
@@ -895,11 +895,11 @@
 
 &ommanager {
 	access-controllers =
-		<&rifsc STM32MP25_RIFSC_OCTOSPI1_ID RIFPROT(STM32MP25_RIFSC_OCTOSPI1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
-		<&rifsc STM32MP25_RIFSC_OCTOSPI2_ID RIFPROT(STM32MP25_RIFSC_OCTOSPI2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
-		<&rifsc STM32MP25_RIFSC_OCTOSPIM_ID RIFPROT(STM32MP25_RIFSC_OCTOSPIM_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
-		<&rifsc STM32MP25_RIFSC_OTFDEC1_ID RIFPROT(STM32MP25_RIFSC_OTFDEC1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
-		<&rifsc STM32MP25_RIFSC_OTFDEC2_ID RIFPROT(STM32MP25_RIFSC_OTFDEC2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>;
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPI1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPI2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPIM_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>;
 
 	st,omm-mux = <0x0>;
 
diff --git a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
index e658506e4..92a0b61f9 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
@@ -895,11 +895,11 @@
 
 &ommanager {
 	access-controllers =
-		<&rifsc STM32MP25_RIFSC_OCTOSPI1_ID RIFPROT(STM32MP25_RIFSC_OCTOSPI1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
-		<&rifsc STM32MP25_RIFSC_OCTOSPI2_ID RIFPROT(STM32MP25_RIFSC_OCTOSPI2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
-		<&rifsc STM32MP25_RIFSC_OCTOSPIM_ID RIFPROT(STM32MP25_RIFSC_OCTOSPIM_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
-		<&rifsc STM32MP25_RIFSC_OTFDEC1_ID RIFPROT(STM32MP25_RIFSC_OTFDEC1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
-		<&rifsc STM32MP25_RIFSC_OTFDEC2_ID RIFPROT(STM32MP25_RIFSC_OTFDEC2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>;
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPI1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPI2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPIM_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>;
 
 	st,omm-mux = <0x0>;
 
diff --git a/core/arch/arm/dts/stm32mp25xc.dtsi b/core/arch/arm/dts/stm32mp25xc.dtsi
index 33930c9f4..c752f7916 100644
--- a/core/arch/arm/dts/stm32mp25xc.dtsi
+++ b/core/arch/arm/dts/stm32mp25xc.dtsi
@@ -11,7 +11,7 @@
 		interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_CRYP1>;
 		resets = <&rcc CRYP1_R>;
-		access-controllers = <&rifsc STM32MP25_RIFSC_CRYP1_ID 0>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_CRYP1_ID>;
 		status = "disabled";
 	};
 
@@ -22,7 +22,7 @@
 		clocks = <&rcc CK_BUS_SAES>, <&rcc CK_BUS_RNG>;
 		clock-names = "bus", "rng";
 		resets = <&rcc SAES_R>;
-		access-controllers = <&rifsc STM32MP25_RIFSC_SAES_ID 0>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_SAES_ID>;
 		status = "disabled";
 	};
 
@@ -33,7 +33,7 @@
 		clocks = <&rcc CK_BUS_PKA>, <&rcc CK_BUS_RNG>;
 		clock-names = "bus", "rng";
 		resets = <&rcc PKA_R>;
-		access-controllers = <&rifsc STM32MP25_RIFSC_PKA_ID 0>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_PKA_ID>;
 		status = "disabled";
 	};
 };
diff --git a/core/arch/arm/dts/stm32mp25xf.dtsi b/core/arch/arm/dts/stm32mp25xf.dtsi
index 476679eb0..d879fd4c6 100644
--- a/core/arch/arm/dts/stm32mp25xf.dtsi
+++ b/core/arch/arm/dts/stm32mp25xf.dtsi
@@ -11,7 +11,7 @@
 		interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_CRYP1>;
 		resets = <&rcc CRYP1_R>;
-		access-controllers = <&rifsc STM32MP25_RIFSC_CRYP1_ID 0>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_CRYP1_ID>;
 		status = "disabled";
 	};
 
@@ -22,7 +22,7 @@
 		clocks = <&rcc CK_BUS_SAES>, <&rcc CK_BUS_RNG>;
 		clock-names = "bus", "rng";
 		resets = <&rcc SAES_R>;
-		access-controllers = <&rifsc STM32MP25_RIFSC_SAES_ID 0>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_SAES_ID>;
 		status = "disabled";
 	};
 
@@ -33,7 +33,7 @@
 		clocks = <&rcc CK_BUS_PKA>,  <&rcc CK_BUS_RNG>;
 		clock-names = "bus", "rng";
 		resets = <&rcc PKA_R>;
-		access-controllers = <&rifsc STM32MP25_RIFSC_PKA_ID 0>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_PKA_ID>;
 		status = "disabled";
 	};
 };
-- 
2.47.2

